### DISCUSSION TITLE
Make `.storage/` more VCS-friendly

### DISCUSSION THREAD

**Discussion by OnFreund** [home-assistant/architecture#901] [General]
With the closure of #370 I'd like to continue the discussion here, as I, and others, still believe this is a needed change.
Reposting the specific steps I proposed:
1. Create a dedicated documentation page for guidelines for VCS tracking.
2. Within that page, recommend which files to exclude (e.g. core.restore_state, mobile_app)
3. Split the auth file by extracting the refresh_tokens section to a new file (and recommend excluding the new file from VCS).
4. Remove icons from the entity registry. Static icons can be configured via customizations, and dynamic icons (based on state) should be handled by the UI.
5. Create a permanent storage mechanism for integrations outside of config entries. This will allow the Spotify integration, for example, to constantly update the token, without updating the config entry.
6. Create a !secret like mechanism for config entries, so that a config entry can be shared without exposing secrets. The auth file can also use that mechanism for credentials. This is required for sharing, not for history tracking, but it's possible that whatever's left after extracting secrets is not useful by itself, so need to consider this carefully.

(I believe 4 was already done, but not 100% sure)

All of these steps I believe will work great together, but some of them can also be implemented independently to make the situation more VCS- friendly.

**Comment by frenck** [comment]
Closing as a duplicate of the linked issue.

Let's not make duplicates.

../Frenck
